pkg/gardenlet/controller/shoot/shoot/reconciler_delete.go

```diff
@@ -223,21 +218,6 @@ func (r *Reconciler) runDeleteShootFlow(ctx context.Context, o *operation.Operat
 			Fn:           flow.TaskFn(botanist.WaitUntilEtcdsReady).DoIf(cleanupShootResources),
 			Dependencies: flow.NewTaskIDs(scaleETCD),
 		})
-		// Redeploy the control plane to make sure all components that depend on the cloud provider secret are restarted
-		// in case it has changed. Also, it's needed for other control plane components like the kube-apiserver or kube-
-		// controller-manager to be updateable due to provider config injection.
-		deployControlPlane = g.Add(flow.Task{
-			Name:         "Deploying Shoot control plane",
-			Fn:           flow.TaskFn(botanist.DeployControlPlane).RetryUntilTimeout(defaultInterval, defaultTimeout).DoIf(cleanupShootResources && controlPlaneDeploymentNeeded).SkipIf(o.Shoot.IsWorkerless),
-			Dependencies: flow.NewTaskIDs(initializeSecretsManagement, deployCloudProviderSecret, ensureShootClusterIdentity),
-		})
-		waitUntilControlPlaneReady = g.Add(flow.Task{
-			Name: "Waiting until Shoot control plane has been reconciled",
-			Fn: flow.TaskFn(func(ctx context.Context) error {
-				return botanist.Shoot.Components.Extensions.ControlPlane.Wait(ctx)
-			}).DoIf(cleanupShootResources && controlPlaneDeploymentNeeded).SkipIf(o.Shoot.IsWorkerless),
-			Dependencies: flow.NewTaskIDs(deployControlPlane),
-		})
```

Comment:
We need to keep these steps in the delete flow in order to ensure that the cloud provider config ist still up-to-date. It is required to delete LBs, volumes etc. from the infrastructure accounts gracefully.

Additionally, you could remove this part of the comment:
```
    Also, it's needed for other control plane components like the kube-apiserver or kube-
// controller-manager to be updateable due to provider config injection.
```